### PR TITLE
DROID-1594 Sets or Collections | Enhancement | Create new view

### PR DESCRIPTION
**Pull request by konstantiniiv**

### CODE REVIEW

File: data/src/main/java/com/anytypeio/anytype/data/auth/repo/block/BlockDataRepository.kt [hunk 1]
```diff
@@ -340,7 +340,7 @@ class BlockDataRepository(
         context: Id,
         target: Id,
         viewer: DVViewer
-    ): Payload = remote.duplicateDataViewViewer(
+    ): Pair<String, Payload> = remote.duplicateDataViewViewer(
```
Comment (uburoiubu) [inline data/src/main/java/com/anytypeio/anytype/data/auth/repo/block/BlockDataRepository.kt:343]: `Pair<Id, Payload>`

File: data/src/main/java/com/anytypeio/anytype/data/auth/repo/block/BlockRemote.kt [hunk 2]
```diff
@@ -125,7 +125,7 @@ interface BlockRemote {
         context: String,
         target: String,
         viewer: DVViewer
-    ): Payload
+    ): Pair<String, Payload>
```
Comment (uburoiubu) [inline data/src/main/java/com/anytypeio/anytype/data/auth/repo/block/BlockRemote.kt:128]: idem

File: domain/src/main/java/com/anytypeio/anytype/domain/block/repo/BlockRepository.kt [hunk 3]
```diff
@@ -167,7 +167,7 @@ interface BlockRepository {
         context: Id,
         target: Id,
         viewer: DVViewer
-    ): Payload
+    ): Pair<String, Payload>
```
Comment (uburoiubu) [inline domain/src/main/java/com/anytypeio/anytype/domain/block/repo/BlockRepository.kt:170]: idem

File: domain/src/main/java/com/anytypeio/anytype/domain/dataview/interactor/DuplicateDataViewViewer.kt [hunk 4]
```diff
@@ -15,9 +15,9 @@ import com.anytypeio.anytype.domain.base.ResultInteractor
 class DuplicateDataViewViewer(
     private val repo: BlockRepository,
     dispatchers: AppCoroutineDispatchers
-) : ResultInteractor<Params, Payload>(dispatchers.io) {
+) : ResultInteractor<Params, Pair<String, Payload>>(dispatchers.io) {
```
Comment (uburoiubu) [inline domain/src/main/java/com/anytypeio/anytype/domain/dataview/interactor/DuplicateDataViewViewer.kt:18]: idem

File: middleware/src/main/java/com/anytypeio/anytype/middleware/block/BlockMiddleware.kt [hunk 5]
```diff
@@ -294,7 +294,7 @@ class BlockMiddleware(
         context: String,
         target: String,
         viewer: DVViewer
-    ): Payload = middleware.blockDataViewViewCreate(
+    ): Pair<String, Payload> = middleware.blockDataViewViewCreate(
```
Comment (uburoiubu) [inline middleware/src/main/java/com/anytypeio/anytype/middleware/block/BlockMiddleware.kt:297]: idem

File: middleware/src/main/java/com/anytypeio/anytype/middleware/interactor/Middleware.kt [hunk 6]
```diff
@@ -297,7 +297,7 @@ class Middleware @Inject constructor(
         context: String,
         target: String,
         viewer: DVViewer
-    ): Payload {
+    ): Pair<String, Payload> {
```
Comment (uburoiubu) [inline middleware/src/main/java/com/anytypeio/anytype/middleware/interactor/Middleware.kt:300]: idem

File: presentation/src/main/java/com/anytypeio/anytype/presentation/sets/EditDataViewViewerViewModel.kt [hunk 7]
```diff
@@ -73,7 +73,7 @@ class EditDataViewViewerViewModel(
                     _toasts.emit("Error while deleting viewer: ${e.localizedMessage}")
                 },
                 onSuccess = {
-                    dispatcher.send(it).also {
+                    dispatcher.send(it.second).also {
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/sets/EditDataViewViewerViewModel.kt:76]: `onSuccess = { (id, payload) -> dispatcher.send(payload) ... }`

**Review by uburoiubu — APPROVED** [review]

File: data/src/main/java/com/anytypeio/anytype/data/auth/repo/block/BlockDataRepository.kt (hunk 1 above)
Comment (konstantiniiv) [reply data/src/main/java/com/anytypeio/anytype/data/auth/repo/block/BlockDataRepository.kt]: fixed

File: presentation/src/main/java/com/anytypeio/anytype/presentation/sets/EditDataViewViewerViewModel.kt (hunk 7 above)
Comment (konstantiniiv) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/sets/EditDataViewViewerViewModel.kt]: done

**Pull request merged**